src/lib.rs

```diff
@@ -176,14 +177,22 @@ pub use poll::Poll;
 // Primitive futures
 mod done;
 mod empty;
+#[cfg(not(feature = "never"))]
 mod failed;
+#[cfg(not(feature = "never"))]
```

Comment:
Can these modules stay even if the `never` feature is enabled? Cargo features in general should be additive in terms of functionality rather than subtractive.